sycl/plugins/level_zero/pi_level_zero.cpp

```diff
@@ -665,12 +695,20 @@ pi_result piDeviceRetain(pi_device Device) {
 
 pi_result piDeviceRelease(pi_device Device) {
   assert(Device);
-
+  pi_platform Platform = Device->Platform;
+  std::lock_guard<std::mutex> Lock(Platform->DeviceCacheMutex);
```

Comment:
please move it closer to the cache use so no locking is done in vain

Reply:
Done